app/uk/gov/hmrc/customs/managesubscription/controllers/ErrorResponses.scala

```diff
@@ -0,0 +1,114 @@
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.customs.managesubscription.controllers
+
+import play.api.http.ContentTypes
+import play.api.libs.json._
+import play.api.mvc.Result
+import play.api.mvc.Results._
+import play.mvc.Http.Status.{BAD_REQUEST, INTERNAL_SERVER_ERROR, NOT_ACCEPTABLE, NOT_FOUND, _}
+
+import scala.xml.NodeSeq
+
+// We can remove
+trait HttpStatusCodeShortDescriptions {
+  // 2XX
+  val OkCode = "OK"
+  val CreatedCode = "CREATED"
+  val AcceptedCode = "ACCEPTED"
+  // 4XX
+  val BadRequestCode = "BAD_REQUEST"
+  val UnauthorizedCode = "UNAUTHORIZED"
+  val NotFoundCode = "NOT_FOUND"
+  val ForbiddenCode = "FORBIDDEN"
+  val MethodNotAllowedCode = "METHOD_NOT_ALLOWED"
+  val NotAcceptableCode = "ACCEPT_HEADER_INVALID"
+  val UnsupportedMediaTypeCode = "UNSUPPORTED_MEDIA_TYPE"
+  // 5XX
+  val InternalServerErrorCode = "INTERNAL_SERVER_ERROR"
+  val NotImplemented = "NOT_IMPLEMENTED"
+  val BadGateway = "BAD_GATEWAY"
+  val ServiceUnavailable = "SERVICE_UNAVAILABLE"
+}
+
+case class ResponseContents(code: String, message: String)
+
+object ResponseContents {
+  implicit val writes: Writes[ResponseContents] = Json.writes[ResponseContents]
+}
+
+case class ErrorResponse(httpStatusCode: Int, errorCode: String, message: String, content: ResponseContents*) extends Error {
+  private lazy val errorContent = JsObject(Seq(
+    "code" -> JsString(errorCode),
+    "message" -> JsString(message))
+  )
+
+  private lazy val responseJson: JsValue = if (content.isEmpty) errorContent else errorContent + ("errors" -> Json.toJson(content))
+
+  lazy val JsonResult: Result = Status(httpStatusCode)(responseJson).as(ContentTypes.JSON)
+  lazy val XmlResult: Result = Status(httpStatusCode)(responseXml).as(ContentTypes.XML)
+
+  private lazy val responseXml: String = "<?xml version='1.0' encoding='UTF-8'?>\n" +
+    <errorResponse>
+      <code>{errorCode}</code>
+      <message>{message}</message>
+      {errors}
+    </errorResponse>
+
+  private val errors =
+    if (content.nonEmpty) {
+      <errors>
+        {content.map(c =>
+        <error>
+          <code>{c.code}</code>
+          <message>{c.message}</message>
+        </error>)}
+      </errors>
+    }
+    else {
+      NodeSeq.Empty
+    }
+
+  def withErrors(contents: ResponseContents*): ErrorResponse = ErrorResponse(this.httpStatusCode, this.errorCode, this.message, contents :_ *)
+}
+
+object ErrorResponse extends HttpStatusCodeShortDescriptions {
+
+  val ErrorUnauthorized = ErrorResponse(UNAUTHORIZED, UnauthorizedCode, "Bearer token is missing or not authorized")
+
+  def errorBadRequest(errorMessage: String, errorCode: String = BadRequestCode): ErrorResponse =
+    ErrorResponse(BAD_REQUEST, errorCode, errorMessage)
+
+  val ErrorGenericBadRequest: ErrorResponse = errorBadRequest("Bad Request")
+
+  val ErrorInvalidPayload: ErrorResponse = errorBadRequest("Invalid payload")
+
+  // NOT USED - NOT SURE IF WE WANT TO KEEP IT
```

Comment:
Would these comments be better prefixed with "TODO" so we can find them later?

Reply:
Yes, good idea